### PR TITLE
Add support for --no-resize option and --keep-awake functionality

### PR DESCRIPTION
**Pull request by sofiaferro**
This pull request introduces two new features to enhance the functionality of the printer script:

1. `--no-resize` option:

- If the image width is smaller than the paper width, it prints without resizing.
- This is particularly useful for printing custom designs or tickets where preserving the layout is crucial.
Usage:

```bash
./print.py --no-resize image.png
```

2. `--keep-awake` functionality:

Adds a process to prevent the printer from entering sleep mode by sending periodic "keep-alive" commands.
This can be activated via the Node.js server by including the `--keep-awake` argument. Here’s the implementation I worked on: [sofiaferro/catprinter-manager](https://github.com/sofiaferro/catprinter-manager).
Usage:

```bash
node server.js --keep-awake
```

Both changes include the necessary updates to the CLI and documentation. These features improve flexibility and address edge cases for printing workflows.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,6 +40,33 @@ options:
...
+# Print without automatic resizing
+./print.py --no-resize test.png
+```
+
+## Printer Keep-Awake
+A keep-awake mechanism can be configured in the manager project to prevent printer sleep mode. This is typically managed externally and can be activated when running the server.
+
+```bash
+# Managed by parent project
+node server.js --keep-awake
```
Comment (rbaron) [inline README.md:69]: I suspect this is a leftover section that mentions "parent project". I would remove this from the PR.

File: catprinter/img.py [hunk 2]
```diff
@@ -120,19 +120,32 @@ def read_img(
...
-    factor = print_width / width
-    resized = cv2.resize(
-        im,
-        (
-            print_width,
-            int(height * factor)
-        ),
-        interpolation=cv2.INTER_AREA)
 
+    if not no_resize:
```
Comment (rbaron) [inline catprinter/img.py]: Nit: I have some trouble reading double negatives like this. Maybe we can invert the if-else and possibly some nesting with something like:

```
if no_resize and width <= print_width: # Print as is.
  ...
else: # Resize to fit width. Preserves aspect ratio.
  ...
```

What do you think?

File: dummy-print.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
+import asyncio
```
Comment (rbaron) [inline dummy-print.py:1]: This `dummy-print.py` shares a lot of code with `print.py`. I would suggest adding a `--send-keepalive` to `print.py` instead. It will make maintenance easier.

Maybe we can also mention it in the README.md instead of the nodejs application there. Something like "To prevent the printer from sleeping, call `print.py --send-keepalive` every x seconds". I think that would be helpful to others.

**Review by rbaron — COMMENTED** [review]
Thanks @sofiaferro. I was very happy to see catprinter used in your art installation!

File: dummy-print.py (hunk 3 above)
Comment (sofiaferro) [reply dummy-print.py:1]: You're right that `dummy-print.py` shares a lot of code with `ble.py`. The keep-alive functionality currently requires a parent process to keep the printer awake between prints, which isn't something handled directly by `print.py` or `ble.py`.

Alternatively, we could move this functionality to `ble.py` to avoid duplication, or let `dummy-print.py` exist in a fork if it's too specific for the main branch. This way, the main codebase stays clean while supporting this use case. Let me know your thoughts!

Comment (rbaron) [reply dummy-print.py:1]: I thought about something like adding a `--send-keepalive` "dummy switch" to `print.py` that only pings the printer with the functionality you already implemented. I think this would be great already.

Then users can choose if/how to periodically run the keepalive loop. For example, some may be happy with a quick and dirty bash loop like:

```bash
$ while true; do python3 print.py --send-keep-alive; sleep 30; done
```

So I think we could get rid of `dummy-print.py` and use the same `print.py` with an additional if-else `send_keepalive`. In your manager, I think it would be enough to change this line to `print.py --send-keepalive` [here](https://github.com/sofiaferro/catprinter-manager/blob/db27685cfa8786887218ae337932d92ae054d1c5/services/scheduleKeepAwake.js#L8).

I would still link your [catprinter-manager](https://github.com/sofiaferro/catprinter-manager) in the README because it has nice features for advanced users (queue, better keepalive etc), and we get to keep this one very simple. What do you think?
